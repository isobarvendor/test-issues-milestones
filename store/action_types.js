export const LOGIN= `login`;
export const GET_TOKEN= `getToken`;
export const SIGNUP= `signup`;
export const GET_ACCOUNT= `getAccount`;

//ngps
export const GET_LIST_PRIZE= `getListPrize`;
export const GET_LIST_WALLET= `getListWallet`;
export const CHECK_ATTEMPT= `checkAttempt`;
export const SUBMIT_FORM= `submitForm`;
export const UPLOAD_FILE= `uploadFile`;
export const DELETE_FILE= `deleteFile`;
export const REDEEM_PRIZE=`redeemPrize`;
export const GET_MY_PRIZE=`getMyPrize`;
export const GET_PHONE = `getPhone`;
export const GET_USER_DATA = `getUserData`

//cms
export const FETCH_CMS_DATA= `fethCMSData`;
export const GET_LIST_WINNERS =`getListWinners`;
