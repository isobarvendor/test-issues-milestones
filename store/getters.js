import deepClone from 'deep-clone';

export default {
    getLoginAccount(state) {
        return deepClone(state.login);
    },
    getCMSContent(state) {
        return deepClone(state.CMSContent);
    },
    getCMSConfig(state) {
      return deepClone(state.config);
    },
    getListPrize(state) {
      return deepClone(state.listPrize);
    }
}
