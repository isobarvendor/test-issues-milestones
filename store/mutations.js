

//shared store are concatenated to the central store
//TO-DO: manage the shared store seperately by module
export default {

  SET_LOGIN_ACCOUNT:(state, data)=>{
        state.login = data;
    },

  SET_TOKEN:(state, data)=>{
      state.token = data;
  },
  SET_ERROR:(state, data)=>{
    state.errorLogin = data;
},
  SET_LIST_PRIZE:(state, data)=>{
    state.listPrize = data;
  },
  SET_LIST_WALLET:(state, data)=>{
    state.listWallet = data;
  },
  SET_FILE_AMAZON:(state, data)=>{
    state.fileUploaded = data;
  },
  SET_CONFIG:(state,data)=>{
    state.config = data
  },
  SET_CMS_CONTENT:(state,data)=>{
    state.CMSContent = data
  }



};
