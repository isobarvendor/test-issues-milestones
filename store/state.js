export default () => ({
    details: {},
    productName: "coke",
    login:null,
    token:null,
    fileUploaded:null,
    listPrize:null,
    errorLogin:null,
    config:null,
    CMSContent:null,
    language:"en",
    redeemPrize:null,
    listWinners:null,
    listWallet:null,
    isCampaignStarted:false,
    goToSignin:false,
    fromBanCity:false,

});
