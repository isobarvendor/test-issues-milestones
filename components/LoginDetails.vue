<template>
<div class="login-details">
  <div class="title-image"><img src="img/icons/coke.png" />
      <p>Sign up with us and gain access to our rewards catalogue where you can redeem prizes.</p>
  </div>
  <div class="continue" v-if="$mq == 'xl'">
      <p>Continue with</p>
      <div class="icons">
          <div v-on:click="redirectTo('line')" v-if="social.line"><span><img src="img/icons/line.png" /></span></div>
          <div v-on:click="redirectTo('facebook')" v-if="social.facebook" ><span><img src="img/icons/fb.png" /></span></div>
          <div v-on:click="redirectTo('google')"  v-if="social.google"><span><img src="img/icons/google.png" /></span></div>
          <div v-on:click="redirectTo('email')"  v-if="social.email"><span><img src="img/icons/email.png" /></span></div>
      </div>
      <p> or </p>
  </div>
  <div class="login-box column center" v-if="$mq == 'sm' || $mq == 'md' || $mq == 'lg'">
    <a class="social-button fb" v-on:click="redirectTo('facebook')"><i class="ic ic-fb" /><span>Continue with Facebook</span></a>
    <a class="social-button line"  v-on:click="redirectTo('line')"  ><i class="ic ic-line" /><span>Continue with LINE</span></a>
    <a class="social-button google"  v-on:click="redirectTo('google')"><i class="ic ic-google" /><span>Continue with Google</span></a>
    <a class="social-button email"  v-on:click="redirectTo('email')"><i class="ic ic-email" /><span>Continue with email</span></a>
  </div>
  <transition name="fade">
    <div class="details" v-if="show">
        <div class="input-container">
            <v-text-field label="Email address" v-validate="'required|email'" name="email"  v-model="login.email"></v-text-field>
            <span class="error-message">{{ errors.first('email') }}</span>
            <v-text-field label="Password"  v-validate="'required'" name="password"   :type="'password'" v-model="login.password"></v-text-field>
            <span class="error-message">{{ errors.first('password') }}</span>
        </div>
        <div class="button-container">
           <div class="error-message" v-if="errorMessage" v-html="errorMessage"></div>
            <a  class="button" v-on:click="userLogin()" >Login</a>
            <p>Forget Password</p>
            <p  v-on:click="userSignUp()" >Sign Up</p>
        </div>

    </div>
  </transition>
</div>
</template>

<script>
import { LOGIN } from '@/store/action_types';
export default {
  name: "LoginDetails",
  inject: ['$validator'],
  components: {
  },
  data() {
    return {
        show:true,
        login: {
          email:'',
          password:''
        },
        errorMessage:null
    };
  },
  props: {
    data: null,
    social: null
  },
  computed:{
  },
  methods:{
    testing(){
        return this.show = true;
    },
    redirectTo(source){
      location.href="http://localhost:8080/api/oauth2/authorize/"+source+"?redirect_uri=http://localhost:3000/settoken";
    },
    async userLogin() {
       this.$validator.validateAll().then((valid) => {
         if(valid){
             this.errorMessage=null;
            this.$store.dispatch(LOGIN,this.login)
            .then(function (response) {
              console.log(response);
              this.$router.push('/account');
            })
            .catch((error) =>{
              if(error.response.data.status=='401'){
                this.errorMessage='Please enter the correct email/password';
              }
            })
         }

       });

    },
     userSignUp() {
         this.$router.push('/register');

    }
  },
  beforeMount() {},
  mounted(){
      if(this.$mq == 'sm' || this.$mq == 'md' || this.$mq == 'lg'){
        this.show=false;
       }
  }
};
</script>

<style lang="scss">
.login-details{
    display: flex;
    flex-direction: column;
    align-items: center;
    .error-message{
      color:red;
    }
    .title-image{
        img{
            width: 230px;
            margin: auto;
            display: table;
            margin-bottom: 25px;
        }
        width: 50%;
        margin-bottom: 20px;
        text-align: center;
        line-height: 24px;
    }
    .continue{
        text-align: center;
        p{
            margin-bottom: 15px;
            margin-top: 15px;
        }
        .icons{
            width: 350px;
            display: flex;
            justify-content: space-evenly;
        }
    }
    .details{
        width: 50%;
    }
    .v-text-field__slot{
        border-bottom: 1px solid #585858;
    }
    .theme--light.v-label{
        color: #585858;
    }
    .theme--light.v-input input{
        color: #ffffff;
    }
    .button-container{
            margin-top: 40px;
            text-align: center;
            a.button{
                width: 230px;
                display: block;
                margin: auto;
            }
            p{
                margin-top: 40px;
            }
        }

    @media only screen and (max-width: 1199px) {
        .title-image{
            width: 100%;
        }
        .login-box{
            margin-top:25px;
            width:100%;
        }
        .details{
            width: 80%;
            margin-top: 40px;
        }
        .fade-enter-active, .fade-leave-active {
            transition: opacity .5s;
        }
        .fade-enter, .fade-leave-to /* .fade-leave-active below version 2.1.8 */ {
            opacity: 0;
        }
    }
}
</style>
