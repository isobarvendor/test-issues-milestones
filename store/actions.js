
import GeneralAPI from '../api/general';

import {LOGIN, SIGNUP, GET_ACCOUNT} from './action_types';

export default {
  [LOGIN]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.login(data)
        .then(response => {
          commit('SET_TOKEN', response.data.token);
          sessionStorage.token=response.data.token;
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [GET_ACCOUNT]: ({ commit, state, getters }, token) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.getAccount(token)
        .then(response => {

          commit('SET_LOGIN_ACCOUNT', response.data);
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [SIGNUP]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.signup(data)
        .then(messages => {
          //commit('SET_LOGIN_ACCOUNT', messages);
          return resolve(messages);
        })
        .catch(error => {
          console.error(error);
          return reject(error);
        });
      }
    });
  },


  /*[SET_GLOBAL_ERROR_DIALOG]:({commit, state}, status)=>{
    commit('SET_GLOBAL_ERROR_DIALOG', status);
  }
  */
};
