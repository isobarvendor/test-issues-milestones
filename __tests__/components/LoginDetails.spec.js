
import { mount,createLocalVue } from '@vue/test-utils'

import Vuetify from 'vuetify'

import Vuex from 'vuex'

import LoginDetails from "@/components/LoginDetails";

import { LOGIN } from "@/store/action_types";




import actions from '@/store/actions';
import mutations from '@/store/mutations';



Object.defineProperty(window, 'location', {
  writable: true,
  value: { assign: jest.fn() }
});

const redirectTo=(source)=>{
  return "/api/oauth2/authorize/" +
  source +
  "?redirect_uri=" +
  window.location.origin +
  "/settoken";
}

let $mq ="md";
let vuetify;
let localVue, store;


beforeEach(() => {
  vuetify = new Vuetify();
  localVue = createLocalVue()
  localVue.use(Vuex)
  store = new Vuex.Store({
    actions,
    mutations
  })
})


const factory = ($mq) => {
  return mount(LoginDetails, {
    mocks: {
      $mq:$mq
    },
    propsData: {
      data :{
       Title : "Login",
       SubTitle: "Login"
      },
      social:{
        line:true,
        facebook:true,
        google:true,
      }
    },
    vuetify, localVue,store
  });
};

describe("LoginDetails", () => {


  test('should redirect anonymous users to login using facebook mq=md', () => {
    $mq ="md";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    wrapper.find('.fb').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith(redirectTo('facebook'));
  })


  test('should redirect anonymous users to login using line mq=md', () => {
    $mq ="md";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    wrapper.find('.line').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith(redirectTo('line'));
  })

  test('should redirect anonymous users to login using google mq=md', () => {
    $mq ="md";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    wrapper.find('.google').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith(redirectTo('google'));
  })


  test('should redirect anonymous users to login using facebook mq=sm', () => {
    $mq ="sm";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    wrapper.find('.fb').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith(redirectTo('facebook'));
  })


  test('should redirect anonymous users to login using line mq=sm', () => {
    $mq ="sm";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    wrapper.find('.line').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith(redirectTo('line'));
  })

  test('should redirect anonymous users to login using google mq=sm', () => {
    $mq ="sm";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    wrapper.find('.google').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith(redirectTo('google'));
  })


  test('should redirect anonymous users to login using facebook mq=xl', () => {
    $mq ="xl";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    wrapper.find('.fb').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith(redirectTo('facebook'));
  })


  test('should redirect anonymous users to login using line mq=xl', () => {
    $mq ="xl";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    wrapper.find('.line').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith(redirectTo('line'));
  })

  test('should redirect anonymous users to login using google mq=xl', () => {
    $mq ="xl";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    wrapper.find('.google').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith(redirectTo('google'));
  })
  test('should redirect anonymous users to register page', async() => {
    $mq ="xl";
    const wrapper = factory($mq);
    window.location.assign= jest.fn() // Create a spy
    await wrapper.find('.signUp').trigger('click');
    expect(window.location.assign).toHaveBeenCalledWith("/register");
  })

  test("user type login and click login", async() => {
    const wrapper = factory();
    const email = 'ronald.pranata@isobar.com';
    const password = 'password';
    const commit = jest.fn()
    const dispatch = jest.fn()

    wrapper.find('#email').setValue(email);
    wrapper.find('#password').setValue(password);

    expect(wrapper.vm.login.email).toBe(email);
    expect(wrapper.vm.login.password).toBe(password);

    await wrapper.find('.login').trigger('click');
    wrapper.vm.$validator.validate();
    expect(wrapper.vm.$validator.errors.any()).toBe(false);


   // const result = actions[LOGIN]({ commit, dispatch });

    //expect(dispatch).toHaveBeenCalled();


  });

  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });


  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
