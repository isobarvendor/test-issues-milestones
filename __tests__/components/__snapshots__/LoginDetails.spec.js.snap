// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginDetails renders properly 1`] = `
<div class="login-details">
  <div class="title-image"><img src="img/icons/coke.png">
    <p>
      Login
    </p>
  </div>
  <!---->
  <!---->
  <transition-stub name="fade">
    <div class="details">
      <div class="input-container">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="email" class="v-label theme--light" style="left: 0px; position: absolute;">Email address</label><input name="email" id="email" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div> <span class="error-message"></span>
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="password" class="v-label theme--light" style="left: 0px; position: absolute;">Password</label><input name="password" id="password" type="password"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div> <span class="error-message"></span>
      </div>
      <div class="button-container">
        <!---->
        <div class="btn-area"><a class="button login">Login</a></div>
        <p class="signUp">Sign Up</p>
      </div>
    </div>
  </transition-stub>
</div>
`;
